.github/workflows/CI.yml

```diff
@@ -0,0 +1,60 @@
+name: CI
+on:
+  push:
+    branches:
+      - master
+    tags: '*'
+  pull_request:
+concurrency:
+  # Skip intermediate builds: always.
+  # Cancel intermediate builds: only if it is a pull request build.
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ startsWith(github.ref, 'refs/pull/') }}
+jobs:
+  test:
+    name: Julia ${{ matrix.version }} - ${{ matrix.os }} - ${{ matrix.arch }} - ${{ github.event_name }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        version:
+          - '1.0'
+          - '1.7'
+          - 'nightly'
+        os:
+          - ubuntu-latest
```

Comment:
Also, add macOS and windows here.

To reduce CI time and save resources, you can use something like [this](https://github.com/JuliaImages/ImageCore.jl/blob/9e64fecfdc6147753ebdd6b918d0bb435a89fb7f/.github/workflows/UnitTest.yml#L16-L29)

The idea is to

- do 32bit test on Ubuntu only since linux is the cheapest and fastest resource.
- for windows and macOS, only test latest stable `1` version. In most cases, if this package doesn't directly depend on `_jll` artifacts, the LTS version test on ubuntu could be sufficient to catch the compatibility issues.